### PR TITLE
chatzone-desktop: init at 5.2.1

### PR DESCRIPTION
**Pull request by JarvisCraft**
## Description of changes

[Chatzone(-desktop)](https://apps.o3team.ru/?mainNav=Chatzone) is a corporate fork of Mattermost(-desktop) with some performance improvements. It is maintained by [Ozon Tech](https://ozon.tech/) and is available as an `x86_64` AppImage.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
- [x] Tested, as applicable:
  - *not applicable*
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - *none required*
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ch/chatzone-desktop/update.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+#!nix-shell -i bash -p curl common-updater-scripts
+
+set -eu -o pipefail
+
+die() {
+    echo "updater.sh: error $1" >&2
+    exit 1
+}
+
+url=$(curl "https://msg.o3.ru/apps/mattermost-desktop-setup-latest-linux" -sLI -o /dev/null -w '%{url_effective}')
```
[the remark this answers is not in the document]
Comment (ythosa) [reply pkgs/by-name/ch/chatzone-desktop/update.sh:11]: I think we should change the host to `chatzone.o3team.ru`

[the remark this answers is not in the document]
Comment (JarvisCraft) [reply pkgs/by-name/ch/chatzone-desktop/update.sh:11]: It is not the private one, it is the one available directly from [apps.o3team.ru](https://apps.o3team.ru/).

It is public (not hidden behind a VPN) and provides an up-to-date reference to the actual release binary (with the version in its name).

Also note that this is an update script which is only called periodically to upgrade the exact params (version, artifact URL and hash) in the `package.nix` file.

[the remark this answers is not in the document]
Comment (JarvisCraft) [reply pkgs/by-name/ch/chatzone-desktop/update.sh]: > I think we should change the host to chatzone.o3team.ru

That's a neat URL, thanks! I've updates the PR to use it instead.

**Review by yunfachi — COMMENTED** [review]
Also, these two commits should be left. The other commits should be squashed.
```
maintainers: add progrm_jarvis
chatzone-desktop: init at 5.1.10
```

File: pkgs/by-name/ch/chatzone-desktop/package.nix [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+    wrapProgram $out/bin/${pname} \
+        --add-flags "\''${NIXOS_OZONE_WL:+\''${WAYLAND_DISPLAY:+--ozone-platform-hint=auto --enable-features=WaylandWindowDecorations}}"
+
+    runHook postInstall
+  '';
+
+  passthru.updateScript = ./update.sh;
+
+  meta = with lib; {
```
[the remark this answers is not in the document]
Comment (JarvisCraft) [reply pkgs/by-name/ch/chatzone-desktop/package.nix]: Oh, that's a great addition to the style guide!

Fixed in the latest commit

**Comment by JarvisCraft** [comment]
@yunfachi, thanks for your review!

I've removed the use of `with lib;` and reordered+squashed the commits.

File: pkgs/by-name/ch/chatzone-desktop/package.nix [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  ];
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/
+    cp -r bin $out/bin
+
+    wrapProgram $out/bin/${pname} \
+        --add-flags "\''${NIXOS_OZONE_WL:+\''${WAYLAND_DISPLAY:+--ozone-platform-hint=auto --enable-features=WaylandWindowDecorations}}"
```
Comment (SuperSandro2000) [inline pkgs/by-name/ch/chatzone-desktop/package.nix:53]:
```suggestion
    wrapProgram $out/bin/chatzone-desktop \
      --add-flags "\''${NIXOS_OZONE_WL:+\''${WAYLAND_DISPLAY:+--ozone-platform-hint=auto --enable-features=WaylandWindowDecorations}}"
```

File: pkgs/by-name/ch/chatzone-desktop/package.nix [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+  nativeBuildInputs = [ copyDesktopItems makeWrapper ];
+
+  desktopItems = [
+    (makeDesktopItem {
+      inherit icon;
+      name = "chatzone";
+      desktopName = "Chatzone";
+      comment = "Ozon corporate messenger";
+      exec = pname;
```
Comment (SuperSandro2000) [inline pkgs/by-name/ch/chatzone-desktop/package.nix:41]:
```suggestion
      exec = "chatzone-desktop";
```
Using pname like this is an anti pattern

File: pkgs/by-name/ch/chatzone-desktop/package.nix [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+  '';
+in stdenvNoCC.mkDerivation {
+  inherit pname version;
+  src = appimage;
+
+  nativeBuildInputs = [ copyDesktopItems makeWrapper ];
+
+  desktopItems = [
+    (makeDesktopItem {
+      inherit icon;
```
Comment (SuperSandro2000) [inline pkgs/by-name/ch/chatzone-desktop/package.nix:37]: The icon should be installed into the main packages output and only references by file name IIRC.

File: pkgs/by-name/ch/chatzone-desktop/package.nix [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+
+  appimage = appimageTools.wrapType2 {
+    inherit pname version src;
+  };
+  appimageContents = appimageTools.extract {
+    inherit pname version src;
+  };
+  icon = runCommand "chatzone-app-icon.png" {} ''
+    cp ${appimageContents}/app_icon.png $out
+  '';
```
Comment (SuperSandro2000) [inline pkgs/by-name/ch/chatzone-desktop/package.nix:28]: If we copy the icon correctly, we also don't need those two and can inline appimage

**Comment by JarvisCraft** [comment]
@SuperSandro2000 thanks for your review! All fixed
I've also populated more useful desktop entry metadata.

I've also removed the update script for now since the URLs have changed and there is no good alternative for discovering the latest version for now (for history, the last commit with the update script was 39e70d5f2169436df7cc841aba976717deb613b7).

**Comment by JarvisCraft** [comment]
I've also rebased to `master` applying the required `nixfmt` fixes

File: pkgs/by-name/ch/chatzone-desktop/package.nix (hunk 3 above)
Comment (JarvisCraft) [reply pkgs/by-name/ch/chatzone-desktop/package.nix]: Thanks, done

File: pkgs/by-name/ch/chatzone-desktop/package.nix (hunk 4 above)
Comment (JarvisCraft) [reply pkgs/by-name/ch/chatzone-desktop/package.nix]: Thanks for the insight, fixed it

File: pkgs/by-name/ch/chatzone-desktop/package.nix (hunk 6 above)
Comment (JarvisCraft) [reply pkgs/by-name/ch/chatzone-desktop/package.nix]: All done, thanks for the suggestion

File: pkgs/by-name/ch/chatzone-desktop/package.nix (hunk 5 above)
Comment (JarvisCraft) [reply pkgs/by-name/ch/chatzone-desktop/package.nix]: Right, done

**Comment by JarvisCraft** [comment]
Tested the new approach to the icons, it works as intended:

image

**Comment by JarvisCraft** [comment]
Bumped the version to `5.2.1`

File: pkgs/by-name/ch/chatzone-desktop/package.nix [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+  ];
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/
+    cp -r bin $out/bin
+
+    mkdir -p $out/share/${pname}/
+    cp ${appimageContents}/app_icon.png $out/share/${pname}/
```
Comment (SuperSandro2000) [inline pkgs/by-name/ch/chatzone-desktop/package.nix:57]:
```suggestion
    mkdir -p $out/share/chatzone-desktop/
    cp ${appimageContents}/app_icon.png $out/share/chatzone-desktop/
```

File: pkgs/by-name/ch/chatzone-desktop/package.nix [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+    url = "https://cdn1.ozone.ru/s3/chatzone-clients/ci/31072024-1/407/chatzone-desktop-linux-5.2.1.AppImage";
+    hash = "sha256-IXn7mAY3+2q+/PKcNQbRVW+wbAPMWLUh/DGAic6M898=";
+  };
+
+  appimage = appimageTools.wrapType2 { inherit pname version src; };
+  appimageContents = appimageTools.extract { inherit pname version src; };
+in
+stdenvNoCC.mkDerivation {
+  inherit pname version;
+  src = appimage;
```
Comment (SuperSandro2000) [inline pkgs/by-name/ch/chatzone-desktop/package.nix:25]:
```suggestion
  appimageContents = appimageTools.extract { inherit pname version src; };
in
stdenvNoCC.mkDerivation {
  inherit pname version;

  src = appimageTools.wrapType2 {
    inherit pname version src; 
  };
```

File: pkgs/by-name/ch/chatzone-desktop/package.nix (hunk 7 above)
Comment (JarvisCraft) [reply pkgs/by-name/ch/chatzone-desktop/package.nix]: Thanks, done

File: pkgs/by-name/ch/chatzone-desktop/package.nix (hunk 8 above)
Comment (JarvisCraft) [reply pkgs/by-name/ch/chatzone-desktop/package.nix]: Done :white_check_mark:

**Comment by JarvisCraft** [comment]
@SuperSandro2000, thanks, pushed the fixes

**Comment by JarvisCraft** [comment]
Hi @SuperSandro2000, is there anything else to be done before the PR can be merged?

**Pull request merged**